### DISCUSSION TITLE
How to reset the store to the default values?

### DISCUSSION THREAD

**Discussion by StevenClontz** [joshnuss/svelte-local-storage-store#165] [Q&A]
I want to implement a "Reset app data" button that returns the store to the initial value used in creating the persisted store. Naively, `localstorage.clear()` is insufficient. :-) What's the idiomatic way to accomplish this?

**Comment by StevenClontz** [comment]
One method: I exported constants for those initial values at https://github.com/StevenClontz/LetsSplitUpGang/blob/main/src/stores.ts and simply overwrote the stores using those values like this https://github.com/StevenClontz/LetsSplitUpGang/blob/main/src/components/ResetAppButton.svelte

**Comment by joshnuss** [comment]
Hi @StevenClontz!

I think one way is to set the store back to it's initial value:

```javascript
const store = peristed('mykey', initialValue)

// reset
store.set(initialValue)

// or in a component
$store = initialValue
```

If you want to reset the key in `localStorage`, I'd use `localStorage.removeItem(key)`.

It might be worth adding a `.clear` and `.reset` functions to this store. Will think about it.
